python/pyspark/sql/types.py

```diff
@@ -1209,6 +1219,12 @@ def __new__(self, *args, **kwargs):
         else:
             raise ValueError("No args or kwargs")
 
+    def __init__(self, *args, **kwargs):
+        if hasattr(self, "__fields__") and "count" in self.__fields__:
```

Comment:
Should we check all the names of method? `self.__fields__` is an list, the `in` could be expensive.